src/fann_train.c

```diff
@@ -978,7 +978,7 @@ void fann_update_weights_irpropm(struct fann *ann, unsigned int first_weight, un
 	for(; i != past_end; i++)
 	{
 		prev_step = fann_max(prev_steps[i], (fann_type) 0.0001);	/* prev_step may not be zero because then the training will stop */
-		slope = train_slopes[i];
+		slope = train_slopes[i] - ann->learning_l2_norm * weights[i];
```

Comment:
This only adds the L2 regularization to the irpropm algorithm. The same term should be added to the other `fann_update_weights*()` functions.